flake.nix

```
@@ -50,11 +50,54 @@
           rustToolchain = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
           craneLib = (crane.mkLib pkgs).overrideToolchain rustToolchain;
 
+          # Add nightly Rust toolchain, required for building the rustdocs with combined index landing page.
+          nightlyRustToolchain = pkgs.rust-bin.nightly.latest.default.override {
+            extensions = [ "rust-src" "rust-analyzer" ];
+            targets = [ "wasm32-unknown-unknown" ];
+          };
```

```suggestion
		  # https://github.com/oxalica/rust-overlay/blob/master/README.md#cheat-sheet-common-usage-of-rust-bin
          nightlyRustToolchain = pkgs.rust-bin.selectLatestNightlyWith(toolchain: toolchain.override {
            extensions = [ "rust-src" "rust-analyzer" ];
            targets = [ "wasm32-unknown-unknown" ];
          });
```

we should use this method instead, which will search backwards to avoid flakiness when nightly isn't available on a particular day.